shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java

```diff
@@ -222,6 +223,11 @@ public void setTextInputMethodHandler(@Nullable TextInputMethodHandler textInput
     this.textInputMethodHandler = textInputMethodHandler;
   }
 
+  @VisibleForTesting
+  public TextInputMethodHandler getTextInputMethodHandler() {
```

Comment:
oh, this bit is unfortunate. What about in the test, `getTextInputChannel`, then mockito [spy](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#spy-T-) the channel, destroy then verify that set handler null is called? Then it saves you from having to expose this.

Reply:
To support this `TextInputPlugin` would need to make `textInputChannel` non-final and add an API to replace the channel with the spy.  I'm concerned that this would be uglier than exposing `TextInputChannel.getTextInputMethodHandler`.

Another alternative would be adding a `TextInputPlugin` constructor that allows injecting a mock `TextInputChannel`.

Reply:
Right, I looked at it a bit more, my comment was non-sensical 🤣. Though I think we can make it even easier and clean up slightly as well. See my comment below.